utils/refactor-check-compiles.py

```diff
@@ -85,31 +85,21 @@ def main():
         extra_refactor_args.append('-enable-experimental-concurrency')
         extra_frontend_args.append('-enable-experimental-concurrency')
 
-    # FIXME: `refactor-check-compiles` should generate both `-dump-text` and
-    # `dump-rewritten` from a single `swift-refactor` invocation (SR-14587).
-    dump_text_output = run_cmd([
-        args.swift_refactor,
-        '-dump-text',
```

Comment:
`refactor-check-compiles.py` should forward the output of `-dump-text` to stdout. At the moment you are writing the output of `-dump-rewritten` both to stdout and to `temp_file_path`. I think you should be adding `-rewritten-output-file` to the first invocation of `swift-refactor` and remove the second one, not the other way round.